nxp_driver: Update the fsl_lpi2c.* files.

robert-hh opened the pull request:
`Updated` for MIMXRT1011, -1015, -1021, -1052, -1062 and -1064.
Tested with MIMXRT1011, -1021, -1052 and -1062.
Lacking boards for testing MIMXRT1015 and -1064. But at least
the build is fine.
The problem which was fixed: at transfers without data - just the address - a NAK state was not returned to the calling code. That caused a bus scan() to fail.

robert-hh committed:
nxp_driver: Update the fsl_lpi2c.* files.

Updated for MIMXRT1011, -1015, -1021, -1052, -1062 and -1064.
Tested with MIMXRT1011, -1021, -1052 and -1062.
Lacking boards for testing MIMXRT1015 and -1064. But at least
the build is fine.

hathach commented:
thank you for the update, could I ask whether you bump up the whole sdk or just the i2c files. I would suggest to bump up the whole sdk downloaded from https://mcuxpresso.nxp.com/en/dashboard (need a free account). Since drivers are generally tested together. Current version in this repo is 2.8.5 . There is 2.10.0 available from NXP.

robert-hh commented:
At the moment, I made just a small step and changed the I2C files only, and tested the mimxrt port with the state of the other v2.8.5 files. I downloaded the SDK for some of the MCUs already quite a few months ago and updated them when necessary. So technically I could change all files  and change the mimxrt for breaking changes. But then it might be better to switch to the NXP sdk. Only the sheer size of that (>600MB) is preventing it.

robert-hh commented:
I just checked that by downloading a new built SDK package from NXP. The required I2C change is NOT contained in these packages. It's only part of the mxu-SDK.

robert-hh commented:
I will change that into switching to v2.10 first and add the fsl_lpi2c change on top of that. I only have to test that first, which might take a few days.

robert-hh commented:
I re-opened this PR. Switching to v2.10 looks like a major deal. What I have seen so far, it has a very different clock setting and my project fails for mimxrt1021, mimxrt1052 and mimxrt1062. Fixing that could take a while. So it may be good to have this PR and the PR for the mimxrt1176 as a first step. That would allow me to get rid of a few backporting files.
Edit: The two PRs are independent. The mimxrt1176 includes the I2C changes.
